addons/sale_pdf_quote_builder/controllers/quotation_document.py

```diff
@@ -33,6 +33,7 @@ def upload_document(self, ufile, sale_order_template_id=False):
                     'name': ufile.filename,
                     'mimetype': mimetype,
                     'raw': ufile.read(),
+                    'company_id': request.cookies['cids'],
```

Comment:
It's possible to select multiple companies in the dropdown menu, in which case `request.cookies['cids']` will return multiple ids, resulting in a SQL error:
> `ERROR: invalid input syntax for type integer: "2-1"`

Can't you just do:
```suggestion
                    'company_id': request.env.company.id,
```
instead?

Reply:
Thank you for having seen it !

As `request.env.company.id` refers to user's main company, this will lead to the same result as doing nothing
<img width="251" alt="Capture d’écran 2025-02-06 à 17 05 16" src="https://github.com/user-attachments/assets/7c65f300-bb85-431f-a698-e5265b99926d" />

Reply:
Yeah, apparently `request.env.company` relies on the `allowed_company_ids` context value, which is I believe is present via regular ORM calls, but the upload button works using a native `XMLHttpRequest`, meaning no extra context is present for the controller, and hence no correct `env.company` value to fall back on.